ui/app/components/secret-engine/configure-azure.ts

```diff
@@ -0,0 +1,181 @@
+
+    const { identityTokenAudience, identityTokenTtl, clientSecret, rootPasswordTtl } = this.args.model;
+    const wifAttributesSet = !!identityTokenAudience || !!identityTokenTtl;
+    const azureAttributesSet = !!clientSecret || !!rootPasswordTtl;
+    // if any WIF attributes have been set in the model, set accessType to 'wif'
+    this.accessType = wifAttributesSet ? 'wif' : 'azure';
+    // if there are either WIF or azure attributes, disable user's ability to change accessType
+    this.disableAccessType = wifAttributesSet || azureAttributesSet;
+  }
+
+  @action continueSubmitForm() {
+    this.saveIssuerWarning = '';
+    this.save.perform();
+  }
+
+  // check if the issuer has been changed to show issuer modal
+  // continue saving the configuration
+  submitForm = task(
+    waitFor(async (event: Event) => {
+      event?.preventDefault();
+      this.resetErrors();
+      const { issuerConfig } = this.args;
+
+      if (issuerConfig?.hasDirtyAttributes) {
+        // if the issuer has changed show modal with warning that the config will change
+        // if the modal is shown, the user has to click confirm to continue saving
+        this.saveIssuerWarning = `You are updating the global issuer config. This will overwrite Vault's current issuer ${
+          issuerConfig.queryIssuerError ? 'if it exists ' : ''
+        }and may affect other configurations using this value. Continue?`;
+        // exit task until user confirms
+        return;
+      }
+      await this.save.perform();
+    })
+  );
+
+  save = task(
+    waitFor(async () => {
+      const { model, issuerConfig } = this.args;
+      // check if any of the model attributes have changed
+      // if no changes, transition and notify user
+      // if changes, save the model and notify user
+      // note: "backend" dirties model state so explicity ignore it here
+      const attrChanged = Object.keys(model?.changedAttributes()).some((item) => item !== 'backend');
+      const issuerAttrChanged = issuerConfig?.hasDirtyAttributes;
+      if (!attrChanged && !issuerAttrChanged) {
+        this.flashMessages.info('No changes detected.');
+        this.transition();
+        return;
+      }
+
+      const modelSaved = attrChanged ? await this.saveModel() : false;
+      const issuerSaved = issuerAttrChanged ? await this.updateIssuer() : false;
+
+      if (modelSaved || issuerSaved) {
+        this.transition();
+      } else {
+        // otherwise there was a failure and we should not transition and exit the function
+        return;
+      }
+    })
+  );
+
+  async updateIssuer(): Promise<boolean> {
+    try {
+      await this.args.issuerConfig.save();
+      this.flashMessages.success('Issuer saved successfully');
+      return true;
+    } catch (e) {
+      this.flashMessages.danger(`Issuer was not saved: ${errorMessage(e, 'Check Vault logs for details.')}`);
```

Comment:
Out of curiosity, why are we surfacing this error as a flash instead of at the top of the page. I realize the issuer is sort of a secondary workflow, but flash errors can be hard to follow

Reply:
This is a solid question. This flow was implemented in parallel between Chelsea and myself while she helped me wrap up AWS. For both AWS and Azure it goes as follows:
1. If you modified the issuer, we show a modal and confirm you want to change the issuer value as it's a global change.
2. Regardless of if the change to the issuer fails, we continue saving the model and transition to the details page.
3. Show two flash messages on the transition: was the model saved? was the issuer saved?

So we're using flash messages because we transition instead of staying on the form or modal.

However, if we feel like this deserves a refactor for both AWS and now Azure before we continue this pattern for GCP, I could do something like:
1. If you modified the issuer, then attempt to save that value.
2. If it fails, don't close the issuer modal and display the error on the modal and give the user the option to close it in order to save the modal. 
3. The catch —not a big one—is that I'd have to reset the issuer value after they close the modal so that when they attempt to save the form, we don't surface the modal again and they can just save the model.

Thoughts?

Reply:
Based on what we discussed in standup today - I _think_ we landed on:

1. Clicking a button to "edit issuer" inline and separate from the parent form. This opens a modal and includes all of the issuer save functionality (sort of like our inline create new search select workflow that includes a form within a form)
   - Clicking "Cancel" reverts any issuer changes.
3. Reserve the form for just the model params, displaying an error banner if it fails which is our standard pattern

Reply:
I'm going to go ahead and make a JIRA ticket for updating the workflow to what you listed above (that's also how I remember it). I'm moving this work out of this ticket because AWS' workflow also needs to be updated to match.